### ISSUE TITLE
error: inflate: data stream error (invalid stored block lengths) error: corrupt loose object

### ISSUE THREAD

**Issue by Merikona** [desktop/desktop#17832]
### The problem

My work laptop shut down due to motherboard issues while coding a while back, and I had corrupted files after rebooting. I made a commit to my repository around that time (possibly after rebooting), and now, I cannot make further commits. The error message I see is "error: inflate: data stream error (invalid stored block lengths)
error: corrupt loose object '2a8d5a0efc1d84d40c1253920eeff8f608d9de93'
fatal: unable to read 2a8d5a0efc1d84d40c1253920eeff8f608d9de93." 

For the commit following that one, error: inflate: data stream error (invalid stored block lengths)
error: corrupt loose object '2a8d5a0efc1d84d40c1253920eeff8f608d9de93'
fatal: unable to read 2a8d5a0efc1d84d40c1253920eeff8f608d9de93
:100644 100644 78b5949 39de88a F1.png:100644 000000 2150e39 0000000 F2.asv:100644 100644 2150e39 d72010a F3.m:100644 100644 2a8d5a0 ed20078 F4.png:100644 100644 12ebea8 b2db0ab F5.png:100644 100644 dea6ff4 755a193 F6.png:100644 100644 87a15a5 57d48a6 F7.png:100644 100644 52cb843 f8b5803 F8.png:100644 100644 8cd1044 e15268d F9.png:100644 100644 4957e27 4052841 F10.png:100644 100644 7d12c10 768896d F11.png:100644 100644 db1e50d 75d5b0f F12.png:100644 100644 9248e1c 781accd Folder/F13.m:100644 100644 0257c60 0257c60 Folder/Folder/F14.m:000000 100644 0000000 0257c60 Folder/F15.m:100644 100644 0257c60 6761908 Folder/F16.m:100644 100644 0d10f38 1e6ddfa F17.m:000000 100644 0000000 3c2f863 F18.m

Pushing origin since then just gives me the error, "The remote disconnected. Check your Internet connection and try again." I've deleted the corrupted files yet it does not help. 

I've been manually backing my code since the error, and would really like to return to using Github Desktop. What should I do? 

### Release version

Version 3.3.5 (x64)

### Operating system

Windows 10 Pro

### Steps to reproduce the behavior

N/A. Can provide further info upon request. 

### Log files

[2023-11-03.desktop.production.log](https://github.com/desktop/desktop/files/13588268/2023-11-03.desktop.production.log)

### Screenshots

![image](https://github.com/desktop/desktop/assets/108444328/00fc77e9-de89-4f98-af14-6e3f20f2951f)

**Comment by steveward** [comment]
Thanks for the issue @Merikona. Unfortunately the push is failing due to the corrupt objects:

```
error: inflate: data stream error (invalid stored block lengths)
error: corrupt loose object '2a8d5a0efc1d84d40c1253920eeff8f608d9de93'
fatal: loose object 2a8d5a0efc1d84d40c1253920eeff8f608d9de93 (stored in .git/objects/2a/8d5a0efc1d84d40c1253920eeff8f608d9de93) is corrupt
```

You could try deleting that corrupt object to see if it resolves this issue, but there may be other corruption. Unfortunately the easiest solution may be to clone a fresh copy of the repository and then apply and new changes you've made after the corruption on top of the new repository. 

If you have any questions about how to do that process, let me know.

**Comment by Merikona** [comment]
Hi Steve, thank you for the response. I've deleted the 'loose object' in question, "2a8d5a0efc1d84d40c1253920eeff8f608d9de93," and it seemed to have fixed my issues (at least I can push further commits now). I didn't realize that that was an actual file name (since I didn't create the file) and that it was located in the same folder as my repository. 

However, I would like to learn how to reapply new changes on top of an existing repository. When I go into history, I can see my last un-corrupted commit, and right click it. The options are checkout commit, revert changes in commit, create branch from commit, create tag, cherry-pick commit, copy SHA, copy tag, view on GitHub. Is this where I begin? I've used GitHub Desktop for a while now as a tool to track my changes, but have yet to really utilize its functions and would like to learn how.
